### DISCUSSION TITLE
How can I create a composite primary key and a unique index on a table

### DISCUSSION THREAD

**Discussion by amazingakai** [piccolo-orm/piccolo#943] [Q&A] [answered]
I cannot find any way to create a composite primary key and a unique index on a table, for composite primary keys I found this issue ( #820 ) which isn't resolved yet but I couldn't find anything for indexes can someone tell me whats the best approach for now for implementing a composite primary key and a unique index?

**Comment by sinisaos** [comment, marked as answer]
@AmazingAkai For now, you can do this using empty migration and editing the `run()` method with raw SQL as described in this [link](https://github.com/piccolo-orm/piccolo/discussions/175#discussioncomment-2658605).
